lib-grade.rkt

```diff
@@ -15,40 +17,84 @@
 
 (struct test-timeout test-result () #:transparent)
 
+
+;; This is mostly useful for testing, where if it is #f,
+;; we print to the current port
+(define results-path "/autograder/results/results.json")
+
+(define (set-results-path! pth)
+  (set! results-path pth))
```

Comment:
`results-path`, `autodetect-extension`, and `submission-directory` might be better as [parameters](https://docs.racket-lang.org/guide/parameterize.html)

That does make the caching of `autodetected-file` slightly tricky but you could refetch the file if the extension doesn't match!

Reply:
I hadn't done that because the nested (non-imperative) use seems like it'll be more annoying for what are effectively _configuration_ settings (e.g., having to wrap everything inside `parameterize`). The docs just say that `parameterize` is preferred -- if there _is_ an imperative API that I missed, that would resolve my only hesitation.

Reply:
There is an imperative API! (but of course there's no `@examples` of this in the docs...)

```racket
(define submission-directory (make-parameter "/autograder/submission"))

(submission-directory) ; -> "/autograder/submission"

(submission-directory "/if/you/dont/like/parameterize!")

(submission-directory) ; -> "/if/you/dont/like/parameterize!"
```

Reply:
Thanks! I've just done this (and your other suggestion)